### PR TITLE
fix(#923): ensure there are no duplicated SCM webhooks

### LINKED ISSUE #923: Ensure there are no duplicated SCM webhooks
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

If SCM webhook is duplicated, workflows will be triggered multiple times by one event.

**What you expected to happen**:

There are no duplicated SCM webhooks

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
/assign
/priority p0

### PR DESCRIPTION
**Pull request by supereagle**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #923

Reference to #

**Special notes for your reviewer**:

/cc @zhujian7  @cd1989 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/server/biz/scm/github/github.go [hunk 1]
```diff
@@ -406,29 +406,60 @@ func newClientByToken(token string) *github.Client {
...
-			"content_type": "json",
-		},
-	}
-	owner, name := scm.ParseRepo(repo)
-	_, resp, err := g.client.Repositories.CreateHook(g.ctx, owner, name, &hook)
+	_, err := g.GetWebhook(repo, webhook.URL)
 	if err != nil {
-		if resp.StatusCode == 500 {
-			return cerr.ErrorSCMServerInternalError.Error(g.scmCfg.Server, err)
+		if yes := cerr.ErrorContentNotFound.Derived(err); yes {
```
Comment (zhujian7) [inline pkg/server/biz/scm/github/github.go]: `yes` is useless

File: pkg/server/biz/scm/github/github.go [hunk 2]
```diff
@@ -406,29 +406,60 @@ func newClientByToken(token string) *github.Client {
...
+					return cerr.ErrorSCMServerInternalError.Error(g.scmCfg.Server, hErr)
+				}
+				return hErr
+			}
+			return nil
 		}
 		return err
 	}
+
+	log.Warningf("Webhook already existed: %+v", webhook)
```
Comment (zhujian7) [inline pkg/server/biz/scm/github/github.go]: Should we update it when it already exists?

File: pkg/server/biz/scm/github/github.go (hunk 1 above)
Comment (cd1989) [reply pkg/server/biz/scm/github/github.go]:
```
if ...  !yes {
  return err
}
```
Recommended.

File: pkg/server/biz/scm/github/github.go (hunk 2 above)
Comment (supereagle) [reply pkg/server/biz/scm/github/github.go]: There is no need

File: pkg/server/biz/scm/github/github.go (hunk 1 above)
Comment (supereagle) [reply pkg/server/biz/scm/github/github.go]: Fixed

**Comment by supereagle** [comment]
Have fixed comments, @cd1989  @zhujian7  PTAL

**Pull request merged**